### PR TITLE
ref(*)!: Refactors loader to use the new common package

### PR DESCRIPTION
**Pull request by thomastaylor312**
This removes a bunch of no longer needed code and removes the binary for wasm-pkg-loader in favor of wkg.

### CODE REVIEW

File: crates/wasm-pkg-common/src/registry.rs [hunk 1]
```diff
@@ -151,6 +168,32 @@ impl RegistryMetadata {
                 .map_err(Error::InvalidRegistryMetadata)?,
         ))
     }
+
+    /// Creates a new [`RegistryMetadata`] by attempting to fetch it from a well known path. Only
+    /// errors if an error occurred with the request itself. If the URL was not found, `None` is
+    /// returned
+    pub async fn fetch(domain: &str) -> Result<Option<Self>, Error> {
```
Comment (lann) [inline crates/wasm-pkg-common/src/registry.rs:175]: I'm not opposed to having this as a convenience but `reqwest` is a relatively heavyweight dependency so I'd like to gate it behind a non-default feature e.g. `metadata-client` or something.

File: crates/wasm-pkg-common/Cargo.toml [hunk 2]
```diff
@@ -8,10 +8,17 @@ authors.workspace = true
...
+semver = { workspace = true }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
+tokio = { workspace = true, features = ["rt", "sync"] }
 toml = "0.8.13"
 thiserror = "1.0"
-tracing = "0.1"
\ No newline at end of file
+tracing = "0.1"
+warg-client = { workspace = true }
```
Comment (lann) [inline crates/wasm-pkg-common/Cargo.toml:24]: It looks like some (?) of these new dependencies aren't actually used in this crate.

File: crates/wasm-pkg-loader/src/source/mod.rs [hunk 3]
```diff
@@ -6,7 +6,6 @@ use std::cmp::Ordering;
 
 use crate::{Error, PackageRef, Release};
 
-pub mod local;
```
Comment (lann) [inline crates/wasm-pkg-loader/src/source/mod.rs:9]: Why is this removed?

**Review by lann — COMMENTED** [review]
Partial review

File: crates/wasm-pkg-common/Cargo.toml (hunk 2 above)
Comment (thomastaylor312) [reply crates/wasm-pkg-common/Cargo.toml:24]: Oh yep. I had pulled something in and then backed that out and forgot to remove deps

File: crates/wasm-pkg-common/src/config/mod.rs [hunk 4]
```diff
@@ -142,8 +142,8 @@ impl Config {
     }
 
     /// Sets the default registry.
-    pub fn set_default_registry(&mut self, registry: Option<Registry>) {
-        self.default_registry = registry;
+    pub fn set_default_registry(&mut self, registry: Registry) {
+        self.default_registry = Some(registry);
```
[the remark this answers is not in the document]
Comment (thomastaylor312) [reply crates/wasm-pkg-common/src/config/mod.rs:146]: Oh, I misread it then. I'll revert that

File: crates/wasm-pkg-common/src/registry.rs [hunk 5]
```diff
@@ -64,6 +65,20 @@ pub const REGISTRY_METADATA_PATH: &str = "/.well-known/wasm-pkg/registry.json";
...
+}
+
+/// The configuration returned by an OCI registry
+#[derive(Deserialize)]
+pub struct OciProtocolConfig {
+    pub registry: String,
+    #[serde(rename = "namespacePrefix")]
+    pub namespace_prefix: Option<String>,
+}
+
```
[the remark this answers is not in the document]
Comment (thomastaylor312) [reply crates/wasm-pkg-common/src/registry.rs:81]: So were you thinking this would just be a private struct in the loader crate for now? Want to make sure I put this in the right place

File: crates/wasm-pkg-common/src/registry.rs (hunk 1 above)
Comment (thomastaylor312) [reply crates/wasm-pkg-common/src/registry.rs:175]: Yeah, that makes sense to me. I wondered about some of the heavy deps as well

File: crates/wasm-pkg-loader/src/source/mod.rs (hunk 3 above)
Comment (thomastaylor312) [reply crates/wasm-pkg-loader/src/source/mod.rs:9]: I hadn't seen any examples or anything around that showed this and the loader code so I had assumed this was just an artifact for testing. Can add back

Comment (thomastaylor312) [reply crates/wasm-pkg-loader/src/source/mod.rs:9]: Should the type name for this just be "local" or was there something else you had in mind?

**Comment by thomastaylor312** [comment]
Ok, I've done most of the changes. Tomorrow I'll work on adding local back and moving the configs to the right spot

File: crates/wasm-pkg-common/src/registry.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (lann) [reply crates/wasm-pkg-common/src/registry.rs:80]: Private struct in the loader crate probably makes sense for now. We'll probably dedupe the warg config into `warg-client` and I imagine we'll end up with a `wasm-pkg-oci` crate later.

File: crates/wasm-pkg-loader/src/source/mod.rs (hunk 3 above)
Comment (lann) [reply crates/wasm-pkg-loader/src/source/mod.rs:9]: Yeah this was requested in a registry meeting to make sure we can handle different kinds of sources. I think "local" is fine in this context.

**Comment by lann** [comment]
I pulled a large subset of the changes from this PR into #41 to prevent conflicts with other work I'm doing.

**Comment by thomastaylor312** [comment]
Closing this because most was merged in #41
